packages/experimental/TabList/src/TabList.tsx

```diff
@@ -39,8 +39,46 @@ export const TabList = compose<TabListType>({
     const Slots = useSlots(userProps, (layer) => tabs.state[layer] || userProps[layer]);
 
     // Return the handler to finish render.
-    return () => {
-      return <Slots.root></Slots.root>;
+    return (final: TabListProps, ...children: React.ReactNode[]) => {
+      if (!tabs.state) {
+        return null;
+      }
+
+      const { defaultTabbableElement, isCircularNavigation, ...mergedProps } = mergeProps(tabs.props, final);
+
+      // Populate the tabsItemKeys array.
+      if (children) {
+        // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+        // @ts-ignore - TODO, fix typing error
+        tabs.state.context.tabsItemKeys = React.Children.map(children, (child: React.ReactChild) => {
+          if (React.isValidElement(child)) {
+            // Sets default selected tabItem.
+            // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+            // @ts-ignore - TODO, fix typing error
+            if (tabs.state?.context.selectedKey == null && !child.props.disabled) {
+              // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+              // @ts-ignore - TODO, fix typing error
+              tabs.state.context.selectedKey = child.props.tabKey;
+            }
+            // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+            // @ts-ignore - TODO, fix typing error
+            return child.props.tabKey;
+          }
+        });
+      }
+
+      return (
+        <TabListContext.Provider
+          // Passes in the selected key and a hook function to update the newly selected tabsItem and call the client's onTabsClick callback.
+          value={tabs.state.context}
+        >
+          <Slots.root {...mergedProps}>
+            <Slots.container defaultTabbableElement={defaultTabbableElement} isCircularNavigation={isCircularNavigation}>
```

Comment:
Replace 'isCircularNavigation={isCircularNavigation}' with 'isCircularNavigation'

Reply:
same with defaultTabbableElement

Reply:
No I think we want the current code. If you pass in isCircularNavigation directly then it will always be true. The spec is that the user is able to configure it. defaultTabbableElement is not a boolean prop, so it must take a value. Passing in the name of the prop only works for boolean props and will set it to true.

This would work fine if you're setting the properties on an object of FocusZoneProp type, though.

Reply:
Ah, thanks for the clarification!